src/coprocessor/endpoint.rs

```diff
@@ -245,32 +241,30 @@ impl ReqContext {
     pub fn check_if_outdated(&self) -> Result<()> {
         let now = Instant::now_coarse();
         if self.deadline <= now {
-            return Err(Error::Outdated(self.deadline, now, self.get_scan_tag()));
+            let elapsed = now.duration_since(self.start_time);
+            return Err(Error::Outdated(elapsed, self.get_scan_tag()));
         }
         Ok(())
     }
+
+    pub fn set_deadline(&mut self, request_max_handle_duration: Duration) {
```

Comment:
can you rename this function? Normally when we naming a function as `set_x`, we are expecting to pass `x`'s new value as an argument, which is not the case here. I think `set_max_handle_duration(max_handle_duration)` would be better.